File: solutions/majority-element/majority-element.go
// @Title: Majority Element
// @Author: colinjxc
// @Date: 2021-05-20T02:07:18+08:00
// @URL: https://leetcode-cn.com/problems/majority-element


func majorityElement(nums []int) int {
	res, count := nums[0], 0
	for i := 0; i < len(nums); i++ {
		if count == 0 {
			res, count = nums[i], 1
		} else {
			if nums[i] == res {
				count++
			} else {
				count--
			}
		}
	}
	return res
}

File: solutions/majority-element/majority-element.py
// @Title: Majority Element
// @Author: colinjxc
// @Date: 2021-05-20T02:18:35+08:00
// @URL: https://leetcode-cn.com/problems/majority-element


class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        ans, count = 0, 0
        for x in nums:
            ans = x if count == 0  else ans
            count = count + 1 if ans == x else count - 1

        return ans
